protobuf-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/protobuf/fromconnectdata/ConnectDataToProtobufDataConverter.java

```diff
@@ -18,14 +18,18 @@
  */
 public class ConnectDataToProtobufDataConverter {
 
+    private Map<String, Descriptors.Descriptor> allMessagesByName = null;
+
     public Message convert(
         @NonNull final Descriptors.FileDescriptor fileDescriptor,
         @NonNull final Schema schema,
         @NonNull final Object value) {
         final List<Field> fields = schema.fields();
         final Struct data = (Struct) value;
 
-        Map<String, Descriptors.Descriptor> allMessagesByName = DescriptorTree.parseAllDescriptors(fileDescriptor);
+        if (allMessagesByName == null) {
+            allMessagesByName = DescriptorTree.parseAllDescriptors(fileDescriptor);
```

Comment:
Isn't this cache will be invalid for a different schema object? Are we planning to introduce a schema conversion cache?

Reply:
Yes I agree. Since we are adding caching in the next task, I can add a comment for now and do that later as it will need to share the common customer configuration for caching.